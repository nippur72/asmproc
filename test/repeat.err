
basic_row_0:  BYTE [basic_row_1%256],[basic_row_1/256],[10%256],[10/256],153,32,34,72,69,76,76,79,34,58,153,32,34,87,79,82,76,68,34,0

   20 GOTO 10
BASIC END


BITMAP -------X
BITMAP ------X-
BITMAP -----X--
BITMAP ----X---
BITMAP ---X----
BITMAP --X-----
BITMAP -X------
BITMAP X-------

FLOAT 3.14

FLOAT 0

FLOAT 1

#IFDEF A
   TEXT_IN_IFDEF
#ENDIF


IF CARRY THEN
   THEN_BRANCH
END IF

IF CARRY THEN
   THEN_BRANCH
ELSE
   ELSE_BRANCH
END IF

IF CARRY THEN GOTO SOMELABEL

IF CARRY THEN LDA $03

IF CARRY THEN SOMELABEL

; ===== SUB ... END SUB ====

SUB MYSUB()
   TEXT
   IF CARRY THEN EXIT SUB
   IF CARRY THEN RTS
   TEXT
   EXIT SUB
   TEXT
END SUB

; ===== FOR ... NEXT ====

FOR A=#0 TO #3 STEP #1
   TEXT
   IF CARRY THEN EXIT FOR
   EXIT FOR
NEXT

FOR $03=#1 TO #4 STEP #2
   TEXT
NEXT

; =======================

WHILE CARRY
   TEXT
   IF CARRY THEN EXIT WHILE
   EXIT WHILE
WEND

TEXT
REPEAT_75_START:
   ISTR1
   ISTR2
   IF A>#1 THEN GOTO REPEAT_75_END
   JMP REPEAT_75_END
	BNE REPEAT_75_START§REPEAT_75_END:§
TEXT

REPEAT_83_START:
   TEXT
REPEAT_85_START:
      ISTR1      
	CMP #1§	BCC REPEAT_85_START§REPEAT_85_END:§
   TEXT
	BNE REPEAT_83_START§REPEAT_83_END:§


DO_92_START:
   TEXT
   IF CARRY THEN GOTO DO_92_END
   JMP DO_92_END
	BCC DO_92_START§DO_92_END:§

DO_98_START:
   TEXT
	BCS DO_98_START§DO_98_END:§

DO_102_START:
   TEXT
	BCS DO_102_START§DO_102_END:§
*/
