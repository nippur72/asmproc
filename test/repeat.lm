basic start
   10 print "hello"
      print "world"
   20 goto 10
basic end


bitmap -------X
bitmap ------X-
bitmap -----X--
bitmap ----X---
bitmap ---X----
bitmap --X-----
bitmap -X------
bitmap X-------

float 3.14

float 0

float 1

#IFDEF A
   TEXT_IN_IFDEF
#ENDIF


IF carry THEN
   THEN_BRANCH
END IF

IF carry THEN
   THEN_BRANCH
ELSE
   ELSE_BRANCH
END IF

IF carry THEN GOTO somelabel

IF carry THEN LDA $03

IF carry THEN somelabel

; ===== SUB ... END SUB ====

SUB mysub()
   text
   if carry then exit sub
   if carry then rts
   text
   exit sub
   text
END SUB

; ===== FOR ... NEXT ====

FOR A=#0 TO #3 STEP #1
   TEXT
   IF CARRY THEN EXIT FOR
   EXIT FOR
NEXT

FOR $03=#1 TO #4 STEP #2
   TEXT
NEXT

; =======================

while carry
   text
   if carry then exit while
   exit while
wend

TEXT
repeat
   istr1
   istr2
   if a>#1 then exit repeat
   exit repeat
until zero
TEXT

repeat
   text
   repeat 
      istr1      
   until A>=#1   
   text
until zero


do
   text
   if carry then exit do
   exit do
loop until carry

do
   text
loop while carry

do
   text
loop while carry
*/
