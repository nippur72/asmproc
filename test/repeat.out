TEXT "HELLO ; HELLO" 
TEXT "HELLO // HELLO" 

TEXT 
TEXT 

IF_6_START:
	CPX #10
	BCC IF_6_ELSE
	BEQ IF_6_ELSE
   TEXT
IF_6_ELSE:
IF_6_END:

IF_10_START:
	CPX #10
	BMI IF_10_ELSE
	BEQ IF_10_ELSE
   TEXT
IF_10_ELSE:
IF_10_END:



  
           
                   
                   
             
         


               
               
               
               
               
               
               
               

          

       

       

        
                
      


             
              
      

             
              
    
              
      

                            

                     

                       

                            

           
       
                         
                    
       
           
       
       

                         

                      
       
                         
           
    

                        
       
    

                         

           
       
                           
             
    

    
      
        
        
                           
              
          
    

      
       
          
                 
                 
       
          


  
       
                        
          
                

  
       
                

  
       
                
  
